metadata/client.go

```diff
@@ -183,35 +185,38 @@ func (client *Client) GetFeature(ctx context.Context, feature string) (*Feature,
 }
 
 func (client *Client) GetFeatures(ctx context.Context, features []string) ([]*Feature, error) {
+	logger := logging.GetLoggerFromContext(ctx)
 	stream, err := client.GrpcConn.GetFeatures(ctx)
 	if err != nil {
+		logger.Errorw("Failed to get features", "features", features, "error", err)
 		return nil, err
 	}
 	go func() {
 		for _, feature := range features {
-			stream.Send(&pb.Name{Name: feature})
+			stream.Send(&pb.NameRequest{Name: &pb.Name{Name: feature}, RequestId: logging.GetRequestIDFromContext(ctx)})
 		}
 		err := stream.CloseSend()
 		if err != nil {
-			client.Logger.Errorw("Failed to close send", "Err", err)
+			logger.Errorw("Failed to close send", "error", err)
 		}
 	}()
 	return client.parseFeatureStream(stream)
 }
 
 func (client *Client) GetFeatureVariants(ctx context.Context, ids []NameVariant) ([]*FeatureVariant, error) {
-	logger := client.Logger.With("ids", ids)
+	logger := logging.GetLoggerFromContext(ctx)
```

Comment:
How does this work? We want one logger to exist per request?

Reply:
correct